cpp/src/parquet/encoding.cc

```diff
@@ -2479,7 +2481,12 @@ class DeltaBitPackDecoder : public DecoderImpl, virtual public TypedDecoder<DTyp
       if (ARROW_PREDICT_FALSE(values_current_mini_block_ == 0)) {
         if (ARROW_PREDICT_FALSE(!block_initialized_)) {
           buffer[i++] = last_value_;
-          if (ARROW_PREDICT_FALSE(i == max_values)) break;
+          if (ARROW_PREDICT_FALSE(i == max_values)) {
+            if (i != static_cast<int>(total_value_count_)) {
+              InitBlock();
+            }
+            break;
+          }
```

Comment:
I find this difficult to read, could you add a comment about what this logic is doing?

Reply:
To be honest, the logic is a bit trickey here, I will try to explain it. I'm bad in English, so maybe you can edit it directly if you can explain it better.